.expeditor/scripts/chef-infra-client.spec

```diff
@@ -0,0 +1,89 @@
+# Disable any shell actions, replace them with simply 'true'
+%define __spec_prep_post true
+%define __spec_prep_pre true
+%define __spec_build_post true
+%define __spec_build_pre true
+%define __spec_install_post true
+%define __spec_install_pre true
+%define __spec_clean_post true
+%define __spec_clean_pre true
+
+# Use SHA256 checksums for all files
+%define _binary_filedigest_algorithm 8
+
+%define _binary_payload w1.xzdio
+
+# Disable creation of build-id links
+%define _build_id_links none
+
+Name:           chef-infra-client
+Version:        %{VERSION}
+Release:        1%{?dist}
+Summary:        The full stack of Chef Infra Client
+AutoReqProv: 	no
+BuildRoot: 	    %buildroot
+Prefix: 	    /
+Group: 		    default
+License:        Chef EULA
+URL: 		    https://www.chef.io
+Packager: 	    Chef Software, Inc. <maintainers@chef.io>
+Source0:        %{CHEF_INFRA_TAR}
+Source1:        %{CHEF_MIGRATE_TAR}
+
+%description
+The full stack of Chef Infra Client
+
+%prep
+# noop
+
+%build
+# noop
+
+%install
+# Create the installation directory for the migration tools
+#mkdir -p %{buildroot}/opt/chef
+mkdir -p %{buildroot}/opt/chef/{bin,bundle}
+
+# Untar the migration tools into /opt/chef/bin
+tar -xf %{SOURCE1} -C %{buildroot}/opt/chef/bin
+
+# Copy the chef infra tarball into /opt/chef/bundle
+cp %{SOURCE0} %{buildroot}/opt/chef/bundle/
+
+%files
+/opt/chef
+/opt/chef/bin
+/opt/chef/bin/*
+/opt/chef/bundle
+/opt/chef/bundle/*
+
+%post
+
+# Determine if --fresh_install needs to be passed based on the existence of the /opt/chef directory
+MIGRATE_CMD="/opt/chef/bin/chef-migrate apply airgap"
+if [ ! -f /opt/chef/bin/chef-client ]; then
```

Comment:
/opt/chef/bin/chef-client may not exist, but chef-client could still be installed on the system and available int he path via workstation. In that case, is 'fresh-install' still correct?

Reply:
The migration tool has an option to uninstall the Omnibus installation during migration. However, I am not sure whether it accounts for the Workstation. Therefore, I am specifically checking the Omnibus installation here

Reply:
Let's instead look at declaring a package-level conflict, so that we can't install infra-19 when chef-workstation 'legacy' is installed.  
That will let us avoid the entire class of problems that arise trying to manually manage conflicting files across packages. 

This can be better supported once workstation itself is released with habitat.

Reply:
Prevents to proceed if chef-workstation is installed: https://github.com/chef/chef/pull/14772/commits/10c403f58b4e4d08c26484cb5b461cae0b7e2983